designs/disruption-controls-by-action.md

```diff
@@ -0,0 +1,188 @@
+# Disruption Controls By Action 
+## User Scenarios 
+1. I as a user would like to ensure upgrades only occur during business hours or stricter windows. I would also like to not restrict the cost savings of consolidation when blocking upgrade via drift.
+2. As a HFT firm, I have specific hours where I need all my compute and I can't tolerate scale down requests for consolidation, but outside of operational hours its fine to scale down. 
+## Desired Behavior + Requirements 
+1. User needs to be able to define an action and a budget/budgets for that action. 
+2. Actions that should be supported should by any disruption actions effected by the current Budgets implementation. (Consolidation, Emptiness, Expiration, Drift) 
+3. Budgets should still support the option to set a given behavior for all disruption actions. If action is unspecified we want to just assume its mean't to apply to all actions. 
+
+## API Design
+### Approach A: Add an action field to disruption Budgets 
```

Comment:
FWIW, we brainstormed this exact change when designing disruption controls, but decided to leave it out for MVP.

Reply:
I can see why there are some complications here. THe biggest in terms of implmentation is we normally track deleting - Most restrictive budget in the pool. This adds complexity for something like per action because then you need to know what is in the process of deleting per action.